### PR TITLE
Added pinned survey groups to permission group

### PR DESCRIPTION
**Pull request by hattwj**
This might be an alternative to the current pull request for pinning survey groups on the nav bar. The benefit of this approach is that each permission group can have their own reports pinned to the nav bar.

An additional bonus is that we don't have to update the settings.yml file and it is tested and working in rails_admin.

### CODE REVIEW

File: app/helpers/permission_groups/role_aggregate_group.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
+##
```
Comment (hattwj) [inline app/helpers/permission_groups/role_aggregate_group.rb]: Might be better in LimeExt namespace, its not only used by PermissionGroup.

**Comment by hattwj** [comment]
Hold open until it has specs

**Comment by wceichelb** [comment]
i find this much preferable to #55, which i'll close now. 

I'll plan to write specs tomorrow.

File: app/controllers/ls_reports_controller.rb [hunk 2]
```diff
@@ -7,9 +7,25 @@ class LsReportsController < ApplicationController
...
+        
+        # collect role aggregates
+        @role_aggregates = @survey_groups.role_aggregates
+
+        # Sort groups alphabetically
+        @survey_groups.sort_by{|group| group.title }
     end
 
+    protected
+
```
Comment (wceichelb) [inline app/controllers/ls_reports_controller.rb]: is `back_to_index` necessary? i don't see it used anywhere else in the app or online. looks like a stub?

Comment (hattwj) [inline app/controllers/ls_reports_controller.rb]: Good catch! If its not used we should drop it.

**Comment by wceichelb** [comment]
after working on specs for this model for a while, i'm led to believe that fixtures won't be able to represent the model relationships well enough to work going forward. 

this is i think because of the static/rigid way that fixtures are read from the `.yml`. the specific case that makes this a problem has to do with the `user.lime_surveys` method.

the series of calls that results is as follows--
```ruby
testing that current_user only sees surveys they're allowed to--

...

user.lime_surveys
    user.role_aggregates
        role_aggregate.ready_for_use?
            role_aggregate.agg_question
                lime_survey.find_question 
```
wherein `role_aggregate.agg_fieldname` is compared to the result of `lime_question.my_column_name`, which is set based on the `lime_question`'s `qid`, its containing `lime_group`'s `gid`, and the group's containing `lime_survey`'s `sid`.

while convoluted, this works. 

the issue arises because rails sets the id of each fixture object randomly on each run of the tests. so while the return value of `lime_question.my_column_name` will change on each test run, i haven't found a way to dynamically set the value of `agg_fieldname` in the `role_aggregates.yml` fixture file. 

i think there may be a way to do it using `ActiveSupport::FixtureSet.identify` at the end of the `.yml` file or in a helper to find the fixture `RoleAggregates` in the test db, get the `LimeSurvey` each is associated with, and then set values such that `ready_for_use?` returns `true`, but that seems like a really hacky way of solving the issue. 

### tl;dr
i'm switching to factories and rewriting the specs.

**Comment by hattwj** [comment]
Thanks a ton for volunteering for the specs, if it starts to make your eyes bleed, let me know.

**Comment by wceichelb** [comment]
mostly all factories now, left one fixture in there just because LimeSurveysLanguagesettings appear to be really static objects.

**Comment by wceichelb** [comment]
alrightly, i believe that's that for tests, which makes this ready to merge?

**Comment by hattwj** [comment]
That sounds good to me.
